### DISCUSSION TITLE
Paragraph styling similar to NSParagraphStyle

### DISCUSSION THREAD

**Discussion by hc-admin** [gonzalezreal/swift-markdown-ui#194] [Q&A] [answered]
Hi there!
First of all thanks for such awesome library!

Is there a way to style a paragraph of text similar to `NSParagraphStyle`? I'd like to modify `firstLineHeadIndent`, `headIndent` and some other properties.

**Comment by gonzalezreal** [comment, marked as answer]
Hi @hc-admin,

There is no `NSParagraphStyle` equivalent in MarkdownUI, but you can achieve most of its functionality using view modifiers. More specifically:

- Use the [`markdownMargin(top:bottom:)`](https://swiftpackageindex.com/gonzalezreal/swift-markdown-ui/2.0.1/documentation/markdownui/markdown/markdownmargin(top:bottom:)-7qhwt) modifier to adjust the paragraph spacing.
- Use the [`padding(_:_:)`](https://developer.apple.com/documentation/swiftui/view/padding(_:_:)) modifier to adjust the head and tail indents. MarkdownUI provides the [`relativePadding(_:length:)`](https://swiftpackageindex.com/gonzalezreal/swift-markdown-ui/2.0.1/documentation/markdownui/markdown/relativepadding(_:length:)) modifier if you want to use [font-relative lengths](https://swiftpackageindex.com/gonzalezreal/swift-markdown-ui/2.0.1/documentation/markdownui/relativesize).
- Use the [`lineSpacing(_:)`](https://developer.apple.com/documentation/swiftui/navigationlink/linespacing(_:)) modifier to adjust the amount of space between lines. Again, MarkdownUI provides the [`relativeLineSpacing(_:)`](https://swiftpackageindex.com/gonzalezreal/swift-markdown-ui/2.0.1/documentation/markdownui/markdown/relativelinespacing(_:)) modifier if you want to use font-relative lengths.
- Finally, use the [`multilinetextalignment(_:)`](https://developer.apple.com/documentation/swiftui/view/multilinetextalignment(_:)) modifier to adjust the text alignment.

Currently, there is no equivalent modifier to set the first line head indent.

To override the current paragraph style, you can use the `markdownBlockStyle(_:body:)` modifier as follows:

```swift
Markdown {
  """
  The sky above the port was the color of television, tuned to a dead channel.

  It was a bright cold day in April, and the clocks were striking thirteen.
  """
}
.markdownBlockStyle(\.paragraph) { label in
  label
    // Adjust the paragraph spacing
    .markdownMargin(top: .zero, bottom: .em(1.5))
    // Adjust the head indent
    .relativePadding(.leading, length: .em(0.5))
    // Adjust the tail indent
    .relativePadding(.trailing, length: .em(1))
    // Adjust the line spacing
    .relativeLineSpacing(.em(0.25))
    // Adjust the text alignment
    .multilineTextAlignment(.center)
}
```

**Reply by hc-admin** [reply]
@gonzalezreal Thanks for the reply! Yes, this would cover most of my cases except `firstLineHeadIndent`. Do you have any ideas how to achieve that in `MarkdownUI` ?

**Reply by gonzalezreal** [reply]
Besides manually prepending each paragraph with a tab, I can't think of other workarounds right now. It depends on the effect you'd like to create.

**Reply by RizwanaDesai** [reply]
@gonzalezreal I wants to add indent in specific paragraph, so i'm trying to change configuration for blockquote, but it didn't work as expected, while it works perfectly for paragraph. [My indent issue link.](https://github.com/gonzalezreal/swift-markdown-ui/issues/347)
